Library/Homebrew/utils/github.rb

```diff
@@ -140,6 +140,7 @@ def open(url, data=nil)
     begin
       if data
         data_tmpfile.write data
+        data_tmpfile.close
```

Comment:
Repeated `close`s are a no-op. (It's not really clear from the documentation, but it worked without error when I tested it, and the source code for `Tempfile#close` shows the implementation is happy with repeated closes.)

`IO#flush` might fix it, too; I'm not sure of the semantics. I'm thinking maybe not necessarily: the [IO#flush doco](http://ruby-doc.org/core-1.9.2/IO.html#method-i-flush) says that "this is Ruby internal buffering only; the OS may buffer the data as well". I'm not sure what that means with respect to visibility to external processes doing an `open`.

Regardless, I prefer `close`, because that makes it clear that you're not going to do any more writing to the file and are completely done with it before the reader starts reading, so you won't be doing interleaved reads and writes. I think that's the only way to be sure you always get a complete and consistent data transfer with plain files (otherwise the reader might do a buffered read ahead of the writer and get a premature EOF). The way we're using it here, `flush` could work too because we never actually do another write to that filehandle. But using `close` ensures we stick with that safe behavior.

Maybe that's not the most idiomatic Ruby way, though.